lib/util/http/index.ts

```diff
@@ -316,6 +316,16 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     });
   }
 
+  async getToml(url: string, options?: Opts): Promise<HttpResponse> {
```

Comment:
Needs a separate test in `lib/util/http/index.spec.ts`

It should also parse the toml and support optional schema validation. please do it like the `getJson`.
It should be a separate PR.

Reply:
Gotcha, I understand now how the JSON side works.  Just to make sure I understand, you want two PRs.

- Implementing `getToml` correctly in lib/util/http/index.spec.ts + tests
- [lib/modules/datasource/custom/index.spec.ts](https://github.com/renovatebot/renovate/pull/28594/files#diff-8476be2d1bd645848654dd3e1cdf357902f3e088c951a65de9d9ea0e761e6570) where we consume `getToml` and also refactor the test suite into smaller ones?

Reply:
I did some more digging and had some more downtime to look into this.

I may be missing something, but I can't do it precisely like `getJson` since the internals of `lib.util.http` delegate to `got` for [parsing the body.](https://github.com/sindresorhus/got/blob/88e623a0d8140e02eef44d784f8d0327118548bc/source/core/response.ts#L140)

The best I can do is to use a more basic type like `text` in the `InternalHttpOptions.responseType`, and then once I have something back from `await this.request<ResT>(url, opts)` I can parse the TOML and then optionally apply the schema.

Still working on the `util.http.index.spec` but it feels like I am finally making progress with my understanding of this area.

It appears some of the other JSON support floating around is for interfacing with other systems / clients.

Reply:
yes, do the parsing in Http class